bms-monitor.js

```diff
@@ -1,18 +1,27 @@
-const {expect} = require('chai');
+const { expect } = require('chai');
 
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
-        console.log('Temperature is out of range!');
-        return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
-        console.log('Charge rate is out of range!');
-        return false;
-    }
-    return true;
+//To check if temp in range,returns true if it passes
+function temperatureInRange(inputTemperature) {
+    return (inputTemperature > 0 && inputTemperature < 45) ? true : false;
+
+}
+//To check if soc in range,returns true if it passes
+function socInRange(inputSoc) {
+    return (inputSoc > 20 && inputSoc < 80) ? true : false;
+}
+//To check if charge_rate in range, returns true if it passes
+function charge_rateInRange(inputCharge_Rate) {
+    return (inputCharge_Rate > 0 && inputCharge_Rate < 0.8) ? true : false;
+}
+
+
+function batteryIsOk(inputTemperature, inputSoc, inputCharge_Rate) {
+    let temperatureValue = temperatureInRange(inputTemperature);
+    let socValue = socInRange(inputSoc);
+    let charge_RateValue = charge_rateInRange(inputCharge_Rate);
+    const parameters = [temperatureValue, socValue, charge_RateValue];
+    let isBatteryOk = parameters.every(value => value);

```

Comment:
Nice approach to aggregate the result with low complexity (the above line 23 doesn't need to change when we add more parameters into line 22)